lib/src/rpc/dial.dart

```diff
@@ -0,0 +1,405 @@
+      ..binaryType = 'arraybuffer'
+      ..id = 0
+      ..negotiated = true
+      ..ordered = true,
+  );
+
+  final negotiationChannel = await peerConnection.createDataChannel(
+    'negotiation',
+    RTCDataChannelInit()
+      ..binaryType = 'arraybuffer'
+      ..id = 1
+      ..negotiated = true
+      ..ordered = true,
+  );
+
+  String? uuid;
+  final didConnect = Completer();
+  final didSetRemoteDesc = Completer();
+
+  // If trickleICE is enabled, set onIceCandidate handler
+  if (!(options.webRtcOptions?.disableTrickleIce ?? config.disableTrickle)) {
+    final offer = await peerConnection.createOffer();
+
+    peerConnection.onIceCandidate = (RTCIceCandidate candidate) async {
+      await didSetRemoteDesc.future;
+      if (candidate.candidate == null) {
+        return;
+      }
+
+      try {
+        final candidateProto = ICECandidate(
+          candidate: candidate.candidate,
+          sdpMid: candidate.sdpMid,
+          sdpmLineIndex: candidate.sdpMLineIndex,
+        );
+        await signalingClient.callUpdate(CallUpdateRequest(uuid: uuid, candidate: candidateProto));
+      } catch (error, st) {
+        _logger.e('Update ICECandidate error', error, st);
+      }
+    };
+
+    await peerConnection.setLocalDescription(offer);
+  }
+
+  negotiationChannel.onMessage = (msg) async {
+    final decodedSDPMap = json.decode(utf8.decode(base64Decode(msg.text))) as Map;
+    final sdp = RTCSessionDescription(
+      decodedSDPMap['sdp'],
+      decodedSDPMap['type'],
+    );
+    final offerCollision = sdp.type == 'offer' && peerConnection.signalingState != RTCSignalingState.RTCSignalingStateStable;
+    if (offerCollision) {
+      return;
+    }
+
+    await peerConnection.setRemoteDescription(sdp);
+
+    if (sdp.type == 'offer') {
+      final mediaConstraints = <String, dynamic>{
+        'audio': true,
+        'video': true,
+      };
+      final answer = await peerConnection.createAnswer(mediaConstraints);
+      await peerConnection.setLocalDescription(answer);
+      final sdpJsonString = _convertSDPtoJsonString(await peerConnection.getLocalDescription());
+      final encodedBase64String = _encodeSDPJsonStringToBase64String(sdpJsonString);
+      await negotiationChannel.send(RTCDataChannelMessage(encodedBase64String));
+    }
+  };
+
+  // Call Signaling Service
+  ResponseStream<CallResponse> callStream;
+  final offer = await peerConnection.createOffer({
+    'mandatory': {
+      'OfferToReceiveAudio': false,
+      'OfferToReceiveVideo': false,
+    }
+  });
+  await peerConnection.setLocalDescription(offer);
+
+  final sdpJsonString = _convertSDPtoJsonString(await peerConnection.getLocalDescription());
+  final encodedSdp = _encodeSDPJsonStringToBase64String(sdpJsonString);
+  try {
+    callStream = signalingClient
+        .call(CallRequest(sdp: encodedSdp, disableTrickle: options.webRtcOptions?.disableTrickleIce ?? config.disableTrickle));
+  } catch (error, st) {
+    //TODO: Add error handling
```

Comment:
This is a different TODO that I didn't catch earlier. But only the `CallUpdate` shouldn't rethrow, since it's not really required for everything to function